### PR TITLE
updating dataset.tags and resuse.tags to be slugified. #293

### PR DESCRIPTION
**Pull request by vinyll**
Not sure where it is actually used and should therefore be also updated.

### CODE REVIEW

File: udata/migrations/2016-01-12-add-tag-slug.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Add the slug property to Tags with a default value based on the name
+ * property.
+ */
+
+ // Taken from underscore.string
+function defaultToWhiteSpace(characters) {
```
Comment (davidbgk) [inline udata/migrations/2016-01-12-add-tag-slug.js]: Maybe we should move these to `migrations/utils.js`?

File: udata/migrations/2016-01-12-add-tag-slug.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  str = str.toLowerCase();
+  return str;
+}
+
+var tags = db.tag.find();
+var count = 0;
+
+tags.forEach(function(tag) {
+  tag.slug = slugify(tag.name);
+  db.tag.save(tag);
```
Comment (davidbgk) [inline udata/migrations/2016-01-12-add-tag-slug.js]: I wonder if re-saving each tag in Python once the new field is added with both `populate_from` and `required` is less prone to bugs.

**Comment by vinyll** [comment]
Dataset and Reuse are the 2 places actually using tags.
They come _dirty_ and are cleaned within the `TagListField`.
The migration is preserved in JS as we should update all _Dataset_ and _Reuse_ instances which would corrupt their datetime.

**Comment by davidbgk** [comment]
r+

**Pull request merged**